core/commands/partials/dedicated_configurator.rb

```diff
@@ -0,0 +1,172 @@
+require_relative '../../services/configuration_generator'
+require 'workers'
+
+# The configurator brings up the configuration for the dedicated machines
+class DedicatedConfigurator
+  def initialize(config, env, logger)
+    @config = config
+    @box_manager = env.box_definitions
+    @repos = env.repos
+    @ui = logger
+    @provider = config.provider
+    @machine_configurator = MachineConfigurator.new(@ui)
+    @attempts = env.attempts&.to_i || 5
+    @recreate_nodes = env.recreate
+    NetworkSettings.from_file(config.network_settings_file).match(
+      { ok: ->(result) { @network_settings = result },
+        error: ->(_result) { @network_settings = NetworkSettings.new } }
+    )
+    @threads_count = @config.node_names.length
+    Workers.pool.resize(@threads_count)
+  end
+
+  # Brings up nodes
+  def up
+    nodes = @config.node_names
+    configure_status = configure_nodes(nodes)
+    return configure_status if configure_status.error?
+
+    @network_settings.store_network_configuration(@config)
+    Result.ok('')
+  end
+
+  def configure_node(node, logger)
+    result = nil
+    @attempts.times do |attempt|
+      @ui.info("Configure node #{node}. Attempt #{attempt + 1}.")
+      node_network = @network_settings.node_settings(node)
+      result = connect(node_network, node).and_then do
+        NetworkChecker.resources_available?(@machine_configurator, node_network, logger).and_then do
+          configure_with_chef(node, logger)
+        end
+      end
+
+      break unless try_again?(node, result)
+    end
+    result
+  end
+
+  def try_again?(node, result)
+    if result.success?
+      @ui.info("Node '#{node}' has been configured.")
+      false
+    else
+      @ui.error("Exception during node configuration: #{result.error}")
+      true
+    end
+  end
+
+  # Try connecting to a configured machine
+  def connect(node_network, node)
+    @machine_configurator.run_command(node_network, 'echo connected', @ui)
+    Result.ok('')
+  rescue StandardError
+    Result.error("Failed to establish a connection with #{node}")
+  end
+
+  # Configure single node using the chef-solo respected role
+  def configure_with_chef(node, logger)
+    node_settings = @network_settings.node_settings(node)
+    solo_config = "#{node}-config.json"
+    role_file = ConfigurationGenerator.role_file_name(@config.path, node)
+    unless File.exist?(role_file)
+      @ui.info("Machine '#{node}' should not be configured. Skipping.")
+      return Result.ok('')
+    end
+    extra_files = [
+      [role_file, "roles/#{node}.json"],
+      [ConfigurationGenerator.node_config_file_name(@config.path, node), "configs/#{solo_config}"]
+    ]
+    extra_files.concat(cnf_extra_files(node))
+    @machine_configurator.configure(node_settings, solo_config, logger, extra_files).and_then do
+      node_provisioned?(node)
+    end
+  end
+
+  # Check whether chef have provisioned the server or not
+  def node_provisioned?(node)
```

Comment:
This method is present in Terraform, Vagrant and now Dedicated configurator. This seems odd.